database_size/models/ir_model_size.py

```diff
@@ -0,0 +1,261 @@
+            "attached to other models."
+        ),
+    )
+
+    @api.model
+    def read_group(
+        self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True
+    ):
+        """Enforce that grouped results are ordered.
+
+        Odoo will happily use the grouping field for ordering unless groupby is a
+        list, and as it happens the grouping is usually passed as a list, for
+        example: ['measurement_date:day']
+        """
+        if not orderby and groupby and isinstance(groupby, list | set):
+            field = groupby[0].split(":")[0]
+            orderby = f"{field} desc"
+        return super().read_group(
+            domain,
+            fields,
+            groupby,
+            offset=offset,
+            limit=limit,
+            orderby=orderby,
+            lazy=lazy,
+        )
+
+    @api.depends(
+        "total_table_size",
+        "relations_size",
+        "attachment_size",
+    )
+    def _compute_total_sizes(self):
+        for size in self:
+            size.total_database_size = size.total_table_size + size.relations_size
+            size.total_model_size = size.total_database_size + size.attachment_size
+
+    @api.depends("ir_model_index_size_ids", "ir_model_index_size_ids.size")
+    def _compute_indexes_size(self):
+        for size in self:
+            size.indexes_size = sum(size.ir_model_index_size_ids.mapped("size"))
+
+    @api.depends("ir_model_relation_size_ids", "ir_model_relation_size_ids.size")
+    def _compute_relations_size(self):
+        for size in self:
+            size.relations_size = sum(size.ir_model_relation_size_ids.mapped("size"))
+
+    @staticmethod
+    def _normalize_size(size):
+        """Filter out -1s and compute as MB"""
+        if not size:
+            return 0
+        return int(max(0, size) / (1024 * 1024))
+
+    @api.model
+    def _measure(self):
+        """Create the entries for today's report"""
+        logger = logging.getLogger(__name__)
+        today = fields.Date.context_today(self)
+        # Remove any previous report for the same day
+        self.search([("measurement_date", "=", today)]).unlink()
+        table2model_id = {}
+        for model in self.env.values():
+            if not model._abstract and not model._transient:
+                model_id = self.env["ir.model"]._get_id(model._name)
+                table2model_id[model._table] = model_id
+        model_id2vals = {
+            model_id: {
+                "model_id": model_id,
+                "measurement_date": today,
+                "ir_model_index_size_ids": [],
+                "ir_model_relation_size_ids": [],
+            }
+            for model_id in table2model_id.values()
+        }
+        # Some many2many relation objects are linked explicitely to both models
+        # involved. To prevent counting them double, we will link them to the
+        # largest table. Gather all the related models first.
```

Comment:
Nice catch. However, largest table is not predictive. And this may change over time, which could lead to variations in size. 
What about to take allways the first one ? (first in alphabetical order, or in ID order.)
Also it could simplify algorithm.
not a blocking point, though.

Reply:
The aim of the module is to check for large tables and their growth. Many2many relations are hardly ever symmetric, far from it. Therefore, if the model related table grows, the many2many table grows, and trimming down the large model table will also trim down the many2many table. It would be misleading to attribute the many2many table on any other criteria than size. Indeed, if you start out with 40 partner categories and 10 partners, the linked model will switch but this will only be the case when the table sizes are trivially small. So I would argue against your suggestion.